test/distributed/_shard/test_replicated_tensor.py

```diff
@@ -82,10 +85,11 @@ def test_replicated_tensor_inter_op_tensor(self):
     @skip_if_lt_x_gpu(4)
     @requires_nccl()
     def test_replicated_tensor_inter_op_sharded_tensor(self):
-        local_tensor = torch.ones(3, 3, device=f"cuda:{self.rank}") * 4
-        replica_tensor = ReplicatedTensor(local_tensor)
-
         torch.manual_seed(self.rank)
+
+        local_tensor1 = torch.rand(12, 3, device=f"cuda:{self.rank}") * 4
+        local_tensor2 = torch.ones(12, 3, device=f"cuda:{self.rank}") * 4
```

Comment:
So this is because we need to compare local gathered results and sharded results with the same ReplicatedTensor content, if it's `torch.rand`, the replicated tensor content on each rank will be different so the gathered results will be different and not comparable with the local result